lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1256,7 +1256,11 @@ private function getCommitOrder(array $entityChangeSet = null)
 
                 $joinColumns = reset($assoc['joinColumns']);
 
-                $calc->addDependency($targetClass->name, $class->name, (int)empty($joinColumns['nullable']));
+                $calc->addDependency(
+                    $targetClass->name,
+                    $class->name,
+                    (int) (($joinColumns['nullable'] ?? true) === false)
```

Comment:
I think I wrote this ternary, and honestly I should be ashamed :-\

This should be split.

Reply:
afraid so ;).

I can split it out, but feel free to push to my branch if you have a preferred style.

Reply:
done